edgelet/aziot-edged/src/product_info.rs

```diff
@@ -0,0 +1,194 @@
+use std::convert::AsRef;
+use std::fmt;
+use std::fs;
+use std::io::{self, BufRead};
+use std::path::Path;
+use std::process::{Command, Output};
+
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub(crate) struct ProductInfo {
+    product: Vec<Product>,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Product {
+    name: String,
+    version: String,
+    comment: Option<String>,
+}
+
+#[inline]
+fn read_utf8(v: Vec<u8>) -> io::Result<String> {
```

Comment:
Do we have a common library where we can place these functions? Seems like there are generic enough to be used outside of product_info

Reply:
I think it is best to keep these out of common code since I have not done any efficiency analysis.